docs/react-native-gradle-plugin.md

```diff
@@ -0,0 +1,187 @@
+
+```groovy
+bundleConfig = file(../rn-cli.config.js)
+```
+
+### `bundleAssetName`
+
+This is the name of the bundle file that should be generated. Default is `index.android.bundle`. Can be customized as follow:
+
+```groovy
+bundleAssetName = "MyApplication.android.bundle"
+```
+
+### `entryFile`
+
+The entry file used for bundle generation. The default is to search for `index.android.js` or `index.js`. Can be customized as follow:
+
+```groovy
+entryFile = file("../js/MyApplication.android.js")
+```
+
+### `extraPackagerArgs`
+
+A list of extra flags that will be passed to the `bundle` command. The list of available flags is in [the CLI documentation](https://github.com/react-native-community/cli/blob/main/docs/commands.md#bundle). Default is empty. Can be customized as follows:
+
+```groovy
+extraPackagerArgs = []
+```
+
+### `hermesCommand`
+
+The path to the `hermesc` command (the Hermes Compiler). React Native comes with a version of the Hermes compiler bundled it with, so you generally won't be needing to customize this, by default the plugin will use the correct compiler for your system.
+
+### `hermesFlags`
+
+The list of flags to pass to `hermesc`. By default is just `["-O", "-output-source-map"]`. You can customize it as follows
+
+```groovy
+hermesFlags = ["-O", "-output-source-map"]
+```
+
+## Using Flavors & Build Variants
+
+When building Android apps, you might want to use [custom flavors](https://developer.android.com/studio/build/build-variants#product-flavors) to have different versions of your app starting from the same project.
+
+Please refer to the [official Android guide](https://developer.android.com/studio/build/build-variants) to configure custom build types (like `staging`) or custom flavors (like `full`, `lite`, etc.).
+By default new apps are create with two build types (`debug` and `release`) and no custom flavors.
+
+The combination of all the build types and all the flavors generates a set of **build variants**. For instance for `debug`/`staging`/`release` build types and `full`/`lite` you will have 6 build variants: `fullDebug`, `fullStaging`, `fullRelease` and so on.
+
+If you're using custom variants beyond `debug` and `release`, you need to instruct the React Native Gradle Plugin specifying which of your variants are **debuggable** using the [`debuggableVariants`](#debuggablevariants) configuration as follows:
+
+```diff
+apply plugin: "com.facebook.react"
+
+react {
++ debuggableVariants = ["fullStaging", "fullDebug"]
+}
+```
+
+This is necessary as for `debuggableVariants`, the plugin will skip the JS bundling, so you'll need Metro to run them. So if you list `fullStaging` in the `debuggableVariants`, you won't be able to publish it to a store as it will be missing the bundle.
```

Comment:
```suggestion
This is necessary because the plugin will skip the JS bundling for all the `debuggableVariants`: you'll need Metro to run them. For example, if you list `fullStaging` in the `debuggableVariants`, you won't be able to publish it to a store as it will be missing the bundle.
```